client/html/themes/elegance/aimeos.js

```diff
@@ -4,3 +4,56 @@
  * @license LGPLv3, http://opensource.org/licenses/LGPL-3.0
  * @copyright Aimeos (aimeos.org), 2014
  */
+
+
+
+/**
+ * Hides or shows the catalog filter search reset button
+ * depending on if the input text field is empty or not.
+ */
+AimeosCatalogFilter.hideOrShowCatalogFilterTextSearchResetButton = function() {
+	if ($(this).val() !== '')
+		$(this.nextElementSibling).css('display', 'inline-block');
+	else
+		$(this.nextElementSibling).css('display', 'none');
+};
+
+
+
+/**
+ * Resets the currently catalog filter search.
+ * In the simplest case the entered search term was not submitted
+ * yet, such that the field is just cleared.
+ * The major purpose of this field is to reset the search without
+ * having to manually empty the input field and submit it to show
+ * all items of all categories again.
+ */
+AimeosCatalogFilter.resetCatalogFilterTextSearch = function() {
+//	var catalog_filter_search_input = $('.catalog-filter-search input:nth-of-type(1)');
+//	catalog_filter_search_input.val('');
+	$(this.previousElementSibling).val('');
+	this.parentNode.parentNode.submit();
```

Comment:
Then the user can simply delete the characters - or double click in there.
Because the main use case may be (at least for the HAMAG webpage I used this procedure often) to be able to quickly clear the entire facetted search and not to search further within the criteria.

I agree though that the shop link directly serves the same purpose.